internal/rbd/rbd_util.go

```diff
@@ -1508,6 +1508,49 @@ func (rv *rbdVolume) RepairThickProvision() error {
 	return nil
 }
 
+// DeepCopy creates an independent image (dest) from the source image. This
+// process may take some time when the image is large.
+func (rv *rbdVolume) DeepCopy(dest *rbdVolume) error {
+	opts := librbd.NewRbdImageOptions()
+	defer opts.Destroy()
+
+	// when doing DeepCopy, also flatten the new image
+	err := opts.SetUint64(librbd.ImageOptionFlatten, 1)
+	if err != nil {
+		return err
+	}
+
+	err = dest.openIoctx()
+	if err != nil {
+		return err
+	}
+
+	image, err := rv.open()
+	if err != nil {
+		return err
+	}
+	defer image.Close()
+
+	err = image.DeepCopy(dest.ioctx, dest.RbdImageName, opts)
+	if err != nil {
+		return err
+	}
+
+	// deep-flatten is not supported by all clients, so disable it
+	return dest.DisableDeepFlatten()
```

Comment:
Question: instead of disabling for all clients. can we disable this during mapping time in node server https://github.com/ceph/ceph-csi/blob/devel/internal/rbd/nodeserver.go#L270?

Reply:
I prefer to have a single approach everywhere. Having deep-flatten enabled on some images and not on others sounds confusing.